lib/sqlalchemy/sql/functions.py

```diff
@@ -1797,3 +1801,20 @@ class grouping_sets(GenericFunction[_T]):
     """
     _has_args = True
     inherit_cache = True
+
+
+class string_agg(GenericFunction[_T]):
+    r"""Implement the ``STRING_AGG`` aggregation function
+
+    This function will concatenate non-null values into a string and
+    separate the values by a delimeter.
+
+    The return type of this function is :class:`.String`.
+
+    """
+    type = sqltypes.String()
+    _has_args = True
+    inherit_cache = True
+
+    def __init__(self, sep=",", *args, **kwargs):
```

Comment:
I don't think it has any advantage adding `sep=','` since it will be replaced by the first argument in all cases.

I would just document that the separator is the last argument, and that it may not be needed in some dialects. This would allow implementing also the mysql version, by using as separator the last arg if more than one arg is passed

Reply:
Not 100% sure what you mean about the mysql version

What would happen with the args in the middle (i.e. not the first or last arg), something like `func.string_agg(a, b, c) = group_concat(a, b SEPARATOR c)`? My current implementation would just discard b

Reply:
That would be fine, I think it would error on the db, but that's ok, since it's an user error

In general an error is better that a silent change to the provided args

Reply:
ok, agreed, I have updated accordingly